### PR TITLE
cache: improve efficiency with RangeOption and RangeSeek #1825

### PR DESCRIPTION
**Pull request by ncw**
* All remotes now support RangeOption so remove SeekOption
  * Correct off by one error as RangeOption arguments are inclusive.
  * Use RangeSeek in preference to Seek if available

### CODE REVIEW

**Comment by ncw** [comment]
Can you take a look at this @remusb and see if you think it is OK.

It should make things a bit more efficient!

**Comment by ncw** [comment]
Are you happy for me to merge this @remusb ?

**Comment by remusb** [comment]
Meh, forgot to hit submit on the review. Just one question about the end offset - 1 but it's more a curiosity. It looks good 👍

File: backend/cache/handle.go [hunk 1]
```diff
@@ -389,7 +389,7 @@ func (w *worker) reader(offset, end int64, closeOpen bool) (io.ReadCloser, error
 	r := w.rc
 	if w.rc == nil {
 		r, err = w.r.cacheFs().openRateLimited(func() (io.ReadCloser, error) {
-			return w.r.cachedObject.Object.Open(&fs.SeekOption{Offset: offset}, &fs.RangeOption{Start: offset, End: end})
+			return w.r.cachedObject.Object.Open(&fs.RangeOption{Start: offset, End: end - 1})
```
[the remark this answers is not in the document]
Comment (ncw) [reply backend/cache/handle.go]: The RangeOption is based on the http Range request so it (unlike everything else ever) uses a closed interval [start, end] where end includes the last byte.  What you wrote was assuming it was a half open interval [start, end) - which is traditional now-a-days - so the last byte of end is not included.

**Comment by ncw** [comment]
I'll merge this in a moment... Thanks for taking a look :-)

**Pull request closed without merging**